internal/controller/prefectserver_controller.go

```diff
@@ -203,8 +204,24 @@ func (r *PrefectServerReconciler) reconcileMigrationJob(ctx context.Context, ser
 		condition = conditions.AlreadyExists(objName, errorMessage)
 
 		return &ctrl.Result{}, errors.NewBadRequest(errorMessage)
-	} else if jobNeedsUpdate(&foundMigrationJob.Spec, &desiredMigrationJob.Spec, log) {
-		// TODO: handle replacing the job if something has changed
+	} else if migrationJobNeedsUpdate(&foundMigrationJob.Spec, &desiredMigrationJob.Spec, log) {
+		log.Info("Will replace migration Job", "name", desiredMigrationJob.Name)
+		condition = conditions.NeedsUpdate(objName)
+
+		// deletes the Job, so it can be recreated on the following reconciliation.
+		// k8s Jobs are immutable and need to be recreated if they are to be re-run.
+		if foundMigrationJob.GetDeletionTimestamp() == nil {
+			log.Info("Deleting existing migration Job", "name", foundMigrationJob.Name)
+
+			if err = r.Delete(ctx, foundMigrationJob); err != nil {
```

Comment:
for some odd reason, our test cases never see this `Job` resource get deleted (only soft deleted), so a subsequent create will always fail 🤔